lib/context.js

```diff
@@ -87,7 +87,7 @@ class ContextSession {
       return;
     }
 
-    const json = await this.store.get(externalKey, opts.maxAge, { ctx, rolling: opts.rolling });
+    const json = await this.store.get(externalKey, opts.maxAge, { state: ctx.state, rolling: opts.rolling });
```

Comment:
you can get state from `ctx`, can you just don't change `store.get` and add `ctx` to `store.destroy`?

Reply:
Yep, my first version sends ctx. Then I decided to minimize the pollution of the store methods by sending only the state. I can roll it back to use the ctx.

Each request to my node instance can be to a different db. That's why I need ctx.state.dbName in each store method. 

Here is a simplified example of my store using ctx:
```
const TWO_WEEKS = 14 * 24 * 60 * 60 * 1000;

// Ignore the initialisation of MongoClient at this point 
const MongoClient = new mongodb.MongoClient(mongoURI, mongoOptions);

export get = async (key, maxAge, { rolling, ctx}) => {
    const db = MongoClient.db(ctx.state.dbName);
    const collection = await db.collection('sessions');
    return await collection.findOne({ key });
};

const set = async (key, sess, maxAge, { changed, rolling, ctx}) => {
    const db = MongoClient.db(ctx.state.dbName);
    const collection = await db.collection('sessions');
    const ttl = new Date((typeof maxAge === 'number' ? maxAge : TWO_WEEKS) + Date.now());
    const document = { ...sess, _id: key, ttl };
    await collection.findOneAndUpdate({ _id: key }, { $set: document }, { upsert: true });
};

const destroy = async (key, { ctx}) => {
    const db = MongoClient.db(ctx.state.dbName);
    const collection = await db.collection('sessions');
    await collection.findOneAndDelete({ _id: key });
};
```